### ISSUE TITLE
Expression is incorrect

### ISSUE THREAD

**Issue by fingertap** [sympy/sympy#27230]
In [1]: import sympy as sp

In [2]: x = sp.sympify('-1 + 2', evaluate=False)

In [3]: y = sp.sympify('12 - 3', evaluate=False)

In [4]: x
Out[4]: -1 + 2

In [5]: y
Out[5]: 12 - 1*3

In [6]: z = sp.sympify('x + y').subs({'x': sp.UnevaluatedExpr(x), 'y': sp.UnevaluatedExpr(y)})

In [7]: z
Out[7]: -(1 + 2) + (12 - 1*3)  << should be (-1 + 2) + (12 - 3)

In [8]: eval('-(1 + 2) + (12 - 1*3)')
Out[8]: 6

In [9]: z.simplify()
Out[9]: 10

**Comment by oscarbenjamin** [comment]
The expression is correct. It just apparently displays incorrectly in the str printer:

I'm not sure if the UnevaluatedExpr affects this but we can see this just using `evaluate=False`:
```python
In [9]: z = Add(Add(-1, 2, evaluate=False), Add(12, -3, evaluate=False), evaluate=False)

In [10]: str(z)
Out[10]: '(-1 + 2) + (-3 + 12)'

In [11]: repr(z)
Out[11]: '(-1 + 2) + (-3 + 12)'

In [12]: pretty(z)
Out[12]: '-1 + 2 + -3 + 12'

In [13]: latex(z)
Out[13]: '\\left(-1 + 2\\right) + \\left(-3 + 12\\right)'
```

**Comment by vprayag2005** [comment]
I’m new to open source and interested in working on this issue . Could you please point me to the relevant parts of the codebase or share any tips to get started?

**Comment by oscarbenjamin** [comment]
Actually this issue was fixed in d9680fc5d58d2d4f27d66fa7c7d8503551ace4ee from gh-24030.

The fix was released in SymPy 1.12.
